app/services/users_service.rb

```diff
@@ -4,12 +4,13 @@ class UsersService < BaseService
   def login(email, password)
     result.user = User.find_by(email: email)&.authenticate(password)
 
-    return result.fail!(code: 'incorrect_login_or_password') unless result.user
-    return result.fail!(code: 'incorrect_login_or_password') unless result.user.memberships.active.any?
+    unless result.user && result.user&.memberships&.active&.any?
```

Comment:
Style/SafeNavigation: Use safe navigation (`&.`) instead of checking if an object exists before calling the method.